src/Features/Core/Portable/ExtractMethod/UniqueNameGenerator.cs

```diff
@@ -15,13 +16,19 @@ public UniqueNameGenerator(SemanticModel semanticModel)
             _semanticModel = semanticModel;
         }
 
-        public string CreateUniqueMethodName(SyntaxNode contextNode, string baseName)
+        public string CreateUniqueMethodName(SyntaxNode contextNode, string baseName, ImmutableHashSet<string> localFunctionNames = null)
         {
             Contract.ThrowIfNull(contextNode);
             Contract.ThrowIfNull(baseName);
 
+            if (localFunctionNames == null)
+            {
+                return NameGenerator.GenerateUniqueName(baseName, string.Empty,
+                    n => _semanticModel.LookupSymbols(contextNode.SpanStart, /*container*/null, n).Length == 0);
+            }
+
             return NameGenerator.GenerateUniqueName(baseName, string.Empty,
-               n => _semanticModel.LookupSymbols(contextNode.SpanStart, /*container*/null, n).Length == 0);
+                n => _semanticModel.LookupSymbols(contextNode.SpanStart, /*container*/null, n).Length == 0 && !localFunctionNames.Contains(n));
```

Comment:
i don't get why we need special logic here.    somthing feels wonky as NameGenerator.GenerateUniqueName should have smarts to do this properly.

Reply:
From what I can tell, the LookupSymbols method doesn't seem to take into account local functions when the context node is a MethodDeclarationSyntax.